test/e2e/README.md

```diff
@@ -0,0 +1,21 @@
+# Baremetal Operator end to end tests
+
+The work here is heavily inspired by the Cluster API e2e framework.
+
+The idea is to have fairly flexible framework that allows running tests in different contexts.
+For CI we mainly rely on libvirt, but the tests are meant to be reusable so that they can run also on real hardware.
+This is accomplished by tuning the tests based on a configuration file and flags.
+Examples can be seen in `config`.
+
+In the CI system we set up a pre-configured minikube cluster that is then used for the tests, as seen in [ci-e2e.sh](../../hack/ci-e2e.sh).
+This allows us to have control over the network and have a static configuration for Ironic and the Baremetal Operator.
+The script also creates a VM and configures VBMC to be used as BareMetalHost and BMC respectively.
+
+It is also possible to run the tests with the fixture provider instead of Ironic, but please note that it is quite questionable to call this configuration "end to end".
+This is how to run the e2e test suite with the fixture provider:
+
+```bash
+export E2E_CONF_FILE=${REPO_ROOT}/test/e2e/config/fixture.yaml
+export USE_EXISTING_CLUSTER="false"
+make test-e2e
```

Comment:
Hmm why not give the user a trigger to run text from ci-e2e-sh show he does not need to worry about al the config needed when testing locally ?

Reply:
Developers can run `ci-e2e.sh` if they want to, but I would not expect/require them to do so. It makes quite a lot of assumptions about the environment (e.g. libvirt and docker installed) and uses `sudo` commands. Do you think there would be a benefit to have a `make` target just for running the script?

I am expecting that users will want to set up their own test environments, but then we cannot know what that looks like. For these situations, I would expect that they create their own Kubernetes cluster and installs BMO/Ironic with specific configuration for the environment, including an e2e config file. Then it should be possible to run the tests against the existing cluster.

Reply:
Yeah I agree if you want to keep it generic. My point was clone and run `make test-e2e` does run ci-e2e in capm3 though we have `make e2e-test` that just run the e2e test and keep the env pre-config to the user

Reply:
Got it. I would not want people to blindly clone && make test-e2e so I would prefer to keep it without a make target. If/when we make it portable enough so it works on multiple distros and MacOS then we can add it :slightly_smiling_face: